### DISCUSSION TITLE
machine-learning/train-speedup/deep-learning-compile

### DISCUSSION THREAD

**Discussion by giscus** [luweizheng/lulaoshi.info#72] [Announcements]
# machine-learning/train-speedup/deep-learning-compile

深度学习编译

https://lulaoshi.info/machine-learning/train-speedup/deep-learning-compile.html

**Comment by wenqf11** [comment]
麻烦问一下第一张图中出现graph level 第一个是什么，kernel level的几个分别是什么（除了tvm, autokernel, oneapi,cudnn, xla) 这个图是有出处嘛？

**Reply by luweizheng** [reply]
> 麻烦问一下第一张图中出现graph level 第一个是什么，kernel level的几个分别是什么（除了tvm, autokernel, oneapi,cudnn, xla) 这个图是有出处嘛？

出处：https://github.com/chenzomi12/DeepLearningSystem

Graph Level第一个我也不知道，kernel level：tensor comprehensions,CANN（华为昇腾），mindspore AKG
